src/dotnet/commands/dotnet-restore/Program.cs

```diff
@@ -27,8 +27,16 @@ public static int Run(string[] args)
 
             app.OnExecute(() =>
             {
+                const string nugetXProjWriteTargets = "NUGET_XPROJ_WRITE_TARGETS";
+                bool setXProjWriteTargets = Environment.GetEnvironmentVariable(nugetXProjWriteTargets) == null;
+
                 try
                 {
+                    if (setXProjWriteTargets)
+                    {
```

Comment:
That way someone else higher up can control what the variable should be. If we blindly overwrite the variable, no one else will get a say in what it should be.  It is generally good to allow your callers to override your defaults. They may know better than you. That being said, it is highly unlikely anyone will ever set this variable outside of us. But if they do, we should respect their wishes.

A thing to note is that this is temporary. Restore is moving to 'MSBuild /t:Restore', which will handle installing .props and .targets files from packages appropriately. This is only needed in the interim.